test/new-e2e/utils/e2e/client/stack_initializer.go

```diff
@@ -0,0 +1,69 @@
+package client
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/auto"
+)
+
+type stackInitializer interface {
+	setStack(auth *Authentification, stackResult auto.UpResult) error
+}
+
+type Authentification struct {
+	SSHKey string
+}
+
+func CheckEnvStructValid[Env any]() error {
+	var env Env
+	_, err := getStackInitializers(&env)
+	return err
+}
+
+func CallStackInitializers[Env any](auth *Authentification, env *Env, upResult auto.UpResult) error {
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
At first sight I wondered why a stack has more than one initializer, then realised you have one for each stack element, or stack exposed resource. What about naming it `CallStackResourcesInitializers` ?

Reply:
Or `StackFields`